shell/src/tezedge_state_manager.rs

```diff
@@ -0,0 +1,446 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+//! Module takes care about (initialization, thread starting, ...) one singletons [TezedgeState] and [TezedgeProposer]
+
+use std::collections::HashSet;
+use std::iter::FromIterator;
+use std::net::{IpAddr, SocketAddr};
+use std::sync::mpsc::Receiver;
+use std::sync::{mpsc, Arc};
+use std::time::{Duration, Instant, SystemTime};
```

Comment:
@binier @jurajselep 
do we need SystemTime here?
I remember Juraj, wanted to remove all usage of SystemTime, but I dont remember exactly why

Reply:
`SystemTime` is used only for initial time. Then it is updated using monotonic clock.

If you'll check proposals, now they include `Duration` instead of `Instant`, using that that system time gets updated.

I know why we cant rely on systemtime. In fact it would break my state machine easily because of how not reliable it is...

But I only call `SystemTime::now()` only once, and update it with durations, which are measured by comparing current time in proposer and event time.